dc-cudami-server/dc-cudami-server-business/src/main/java/de/digitalcollections/cudami/server/business/impl/service/identifiable/entity/DigitalObjectServiceImpl.java

```diff
@@ -93,4 +103,30 @@ Identifier getidentifer(DigitalObject digitalObject, String name) {
   public List<DigitalObject> findAllReduced() {
     return ((EntityRepository) repository).findAllReduced(EntityType.DIGITAL_OBJECT);
   }
+
+  @Override
+  public boolean delete(UUID uuid) {
+    // Check for existance. If not given, return false.
+    DigitalObject existingDigitalObject = get(uuid);
+    if (existingDigitalObject == null) {
+      return false;
+    }
+
+    // Remove connection to collections
+    collectionService.removeDigitalObjectFromAllCollections(existingDigitalObject);
```

Comment:
Wie ist denn das hier mit den Foreign Key Constraints, ist durch die Reihenfolge sichergestellt dass es hier nicht zu Problemen, etwa bei zirkulären Abhängigkeiten (die das Cudami-Modell ja glaub erlaubt) kommt?
Vielleicht wäre es sinnvoll diese ganzen Calls in eine Transaktion zu wrappen, dann kann die Datenbank das erkennen und selbst lösen, oder?

Reply:
Eine Transaktionsklammer bei schreibenden Service-Methoden, bei denen mehrere Repositories involviert sind, wäre schon sehr wünschenswert.

Das sollten wir uns allerdings an anderer Stelle ansehen